mailsync

```diff
@@ -1,5 +1,84 @@
 #!/bin/bash
-migsync etc/${1%/} ~/etc
-migsync etc/${1%/} ~/etc
-migsync mail/${1%/} ~/mail
-migsync mail/${1%/} ~/mail
\ No newline at end of file
+
+# check to see if there's an argument
+
+if [ -z "$1" ]; then
+  echo -e 'USAGE: mailsync [domain]'
+  exit 1
+fi 
+
+if [ ! -d ~/migration/migstuff/ ] ; then  
+  mkdir ~/migration/migstuff/ 2>/dev/null
+  mkdir ~/migration/migstuff/logs/ 2>/dev/null
+elif [ ! -d ~/migration/migstuff/logs/ ] ; then 
+  mkdir ~/migration/migstuff/logs/ 2>/dev/null
+  mkdir ~/migration/migstuff/logs/"$1"/ 2>/dev/null
+fi
+
+echo ''
+timestamp=$(date +%s)
+
+# function to count inodes in mail dir for domain 
+
+echo ' ' > ~/migration/migstuff/logs/"$1"/beforesync_"$1"_"$timestamp".txt
+echo ' ' > ~/migration/migstuff/logs/"$1"/aftersync_"$1"_"$timestamp".txt
+
+function mail_Count() {
+  (cd ~/mail/"$1" ; find . -printf "%h\n" | cut -d/ -f-2 | sort | uniq -c | sort -rn)
```

Comment:
why cd here? it makes sense to just pass the path you want in to find instead of using "."
there should also be a check that "~/mail/$1" exists like you do on line 32.  also the $1 doesnt need to be in quotes.

Reply:
reversed inode count changes in mailsync, not counting files for some reason after making requested changes

Reply:
also added sanity check